samples/python/multiview_calibration.py

```
@@ -0,0 +1,220 @@
+                scale = 1000.0 / max(img.shape[0], img.shape[1])
+                if scale < 1.0:
+                    img_detection = cv.resize(img, (int(scale * img.shape[1]), int(scale * img.shape[0])), interpolation=cv.INTER_AREA)
+                    # increase refinement window for the original image resolution
+                    window = (16, 16)
+
+            if pattern_type.lower() == 'checkerboard':
+                ret, corners = cv.findChessboardCorners(img_detection, grid_size, None)
+            elif pattern_type.lower() == 'circles':
+                ret, corners = cv.findCirclesGrid(img_detection, patternSize=grid_size, flags=cv.CALIB_CB_SYMMETRIC_GRID);
+            elif pattern_type.lower() == 'acircles':
+                ret, corners = cv.findCirclesGrid(img_detection, patternSize=grid_size, flags=cv.CALIB_CB_ASYMMETRIC_GRID);
+            else:
+                raise "Calibration pattern is not supported!"
+
+            if ret:
+                if scale < 1.0:
+                    corners /= scale
+                corners2 = cv.cornerSubPix(img, corners, window, (-1,-1), criteria)
+                image_points_camera.append(np.array(corners2, dtype=np.float32).reshape(-1,2))
+            else:
+                image_points_camera.append(np.array([], dtype=np.float32))
+
+        image_points_cameras.append(image_points_camera)
+        image_sizes.append(img_size)
+    calibrateFromPoints(pattern, image_points_cameras, image_sizes, is_fisheye, all_images_names)
+
+def calibrateFromJSON(json_file):
+    assert os.path.exists(json_file)
+    data = json.load(open(json_file, 'r'))
+    for i in range(len(data['image_points'])):
+        for j in range(len(data['image_points'][i])):
+            data['image_points'][i][j] = np.array(data['image_points'][i][j], dtype=np.float32)
+    calibrateFromPoints(np.array(data['object_points'], dtype=np.float32).T, data['image_points'], data['image_sizes'], data['is_fisheye'])
+
+try:
```

Syntax error here. try block is not closed with catch or finally. I propose just to replace try with `if __name__ == '__main__':` and do not hide exceptions, if any.

---

replaced 'try' with if __name__ == '__main__' in new commit here: https://github.com/opencv/opencv/commit/af8359e79347c6d6a5f57e2ead5d3f340fc5fa4e